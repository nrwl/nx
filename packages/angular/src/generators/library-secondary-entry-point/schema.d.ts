import { ProjectConfiguration } from '@nx/devkit';

export interface GeneratorOptions {
  library: string;
  name: string;
  skipModule?: boolean;
  skipFormat?: boolean;
}

export interface NormalizedGeneratorOptions extends GeneratorOptions {
  entryPointDestination: string;
  libraryProject: ProjectConfiguration;
  mainEntryPoint: string;
  secondaryEntryPoint: string;
  skipModule: boolean;
  moduleTypeSeparator: '-' | '.';
}
