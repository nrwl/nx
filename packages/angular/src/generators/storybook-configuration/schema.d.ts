import type { Linter, LinterType } from '@nx/eslint';

export interface StorybookConfigurationOptions {
  configureStaticServe?: boolean;
  generateStories: boolean;
  linter: Linter | LinterType;
  project: string;
  tsConfiguration?: boolean;
  skipFormat?: boolean;
  ignorePaths?: string[];
  interactionTests?: boolean;
}
