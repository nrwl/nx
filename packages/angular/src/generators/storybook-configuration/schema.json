{
  "$schema": "https://json-schema.org/schema",
  "$id": "NxAngularStorybookConfigurationGenerator",
  "title": "Adds Storybook configuration to a project.",
  "description": "Adds Storybook configuration to a project to be able to use and create stories.",
  "cli": "nx",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "aliases": ["name", "projectName"],
      "description": "Project for which to generate Storybook configuration.",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "For which project do you want to generate Storybook configuration?",
      "x-dropdown": "projects",
      "x-priority": "important"
    },
    "interactionTests": {
      "type": "boolean",
      "description": "Set up Storybook interaction tests.",
      "x-prompt": "Do you want to set up Storybook interaction tests?",
      "alias": ["configureTestRunner"],
      "x-priority": "important",
      "default": true
    },
    "generateStories": {
      "type": "boolean",
      "description": "Specifies whether to automatically generate `*.stories.ts` files for components declared in this project or not.",
      "x-prompt": "Automatically generate *.stories.ts files for components declared in this project?",
      "default": true,
      "x-priority": "important"
    },
    "configureStaticServe": {
      "type": "boolean",
      "description": "Specifies whether to configure a static file server target for serving storybook. Helpful for speeding up CI build/test times.",
      "x-prompt": "Configure a static file server for the storybook instance?",
      "default": true,
      "x-priority": "important"
    },
    "linter": {
      "description": "The tool to use for running lint checks.",
      "type": "string",
      "enum": ["eslint", "none"],
      "default": "eslint"
    },
    "tsConfiguration": {
      "type": "boolean",
      "description": "Configure your project with TypeScript. Generate main.ts and preview.ts files, instead of main.js and preview.js.",
      "default": true,
      "x-priority": "important"
    },
    "skipFormat": {
      "description": "Skip formatting files.",
      "type": "boolean",
      "default": false,
      "x-priority": "internal"
    },
    "ignorePaths": {
      "type": "array",
      "description": "Paths to ignore when looking for components.",
      "items": {
        "type": "string",
        "description": "Path to ignore."
      },
      "default": [
        "*.stories.ts,*.stories.tsx,*.stories.js,*.stories.jsx,*.stories.mdx"
      ],
      "examples": [
        "apps/my-app/src/not-stories/**",
        "**/**/src/**/not-stories/**",
        "libs/my-lib/**/*.something.ts",
        "**/**/src/**/*.other.*",
        "libs/my-lib/src/not-stories/**,**/**/src/**/*.other.*,apps/my-app/**/*.something.ts"
      ]
    }
  },
  "additionalProperties": false,
  "required": ["project"],
  "examplesFile": "../../../docs/storybook-configuration-examples.md"
}
