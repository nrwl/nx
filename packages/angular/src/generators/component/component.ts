import type { Tree } from '@nx/devkit';
import { formatFiles, generateFiles, joinPathFragments } from '@nx/devkit';
import { addToNgModule } from '../utils';
import { getInstalledAngularVersionInfo } from '../utils/version-utils';
import {
  exportComponentInEntryPoint,
  findModuleFromOptions,
  normalizeOptions,
  setGeneratorDefaults,
} from './lib';
import type { Schema } from './schema';

export async function componentGenerator(tree: Tree, rawOptions: Schema) {
  const options = await normalizeOptions(tree, rawOptions);

  const { major: angularMajorVersion } = getInstalledAngularVersionInfo(tree);
  generateFiles(
    tree,
    joinPathFragments(__dirname, 'files'),
    options.directory,
    {
      name: options.name,
      fileName: options.fileName,
      symbolName: options.symbolName,
      exportDefault: options.exportDefault,
      style: options.style,
      inlineStyle: options.inlineStyle,
      inlineTemplate: options.inlineTemplate,
      standalone: options.standalone,
      skipSelector: options.skipSelector,
      changeDetection: options.changeDetection,
      viewEncapsulation: options.viewEncapsulation,
      displayBlock: options.displayBlock,
      selector: options.selector,
      // Angular v19 or higher defaults to true, while v18 or lower defaults to false
      setStandalone:
        (angularMajorVersion >= 19 && !options.standalone) ||
        (angularMajorVersion < 19 && options.standalone),
      angularMajorVersion,
      tpl: '',
    }
  );

  if (options.skipTests) {
    const pathToSpecFile = joinPathFragments(
      options.directory,
      `${options.fileName}.spec.ts`
    );

    tree.delete(pathToSpecFile);
  }

  if (options.inlineTemplate) {
    const pathToTemplateFile = joinPathFragments(
      options.directory,
      `${options.fileName}.html`
    );

    tree.delete(pathToTemplateFile);
  }

  if (options.style === 'none' || options.inlineStyle) {
    const pathToStyleFile = joinPathFragments(
      options.directory,
      `${options.fileName}.${options.style}`
    );

    tree.delete(pathToStyleFile);
  }

  if (!options.skipImport && !options.standalone) {
    const modulePath = findModuleFromOptions(
      tree,
      options,
      options.projectRoot
    );
    addToNgModule(
      tree,
      options.directory,
      modulePath,
      options.name,
      options.symbolName,
      options.fileName,
      'declarations',
      true,
      options.export
    );
  }

  exportComponentInEntryPoint(tree, options);
  setGeneratorDefaults(tree, options);

  if (!options.skipFormat) {
    await formatFiles(tree);
  }
}

export default componentGenerator;
