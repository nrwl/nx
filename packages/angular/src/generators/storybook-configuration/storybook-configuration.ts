import {
  formatFiles,
  GeneratorCallback,
  readProjectConfiguration,
  Tree,
} from '@nx/devkit';
import { updateAppEditorTsConfigExcludedFiles } from '../utils/update-app-editor-tsconfig-excluded-files';
import { assertCompatibleStorybookVersion } from './lib/assert-compatible-storybook-version';
import { generateStories } from './lib/generate-stories';
import { generateStorybookConfiguration } from './lib/generate-storybook-configuration';
import type { StorybookConfigurationOptions } from './schema';

export async function storybookConfigurationGenerator(
  tree: Tree,
  options: StorybookConfigurationOptions
): Promise<GeneratorCallback> {
  assertCompatibleStorybookVersion();

  const storybookGeneratorInstallTask = await generateStorybookConfiguration(
    tree,
    {
      ...options,
      interactionTests: options.interactionTests ?? true, // default is true
      tsConfiguration: options.tsConfiguration ?? true, // default is true
    }
  );

  if (options.generateStories) {
    await generateStories(tree, {
      ...options,
      interactionTests: options.interactionTests ?? true,
      skipFormat: true,
    });
  }

  const project = readProjectConfiguration(tree, options.project);
  if (project.projectType === 'application') {
    updateAppEditorTsConfigExcludedFiles(tree, project);
  }

  if (!options.skipFormat) {
    await formatFiles(tree);
  }

  return storybookGeneratorInstallTask;
}

export default storybookConfigurationGenerator;
