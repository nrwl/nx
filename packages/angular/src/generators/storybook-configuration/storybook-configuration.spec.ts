import type { Tree } from '@nx/devkit';
import { readJson, writeJson } from '@nx/devkit';
import { Linter } from '@nx/eslint/src/generators/utils/linter';
import { componentGenerator } from '../component/component';
import { librarySecondaryEntryPointGenerator } from '../library-secondary-entry-point/library-secondary-entry-point';
import {
  createStorybookTestWorkspaceForLib,
  generateTestApplication,
} from '../utils/testing';
import type { StorybookConfigurationOptions } from './schema';
import { storybookConfigurationGenerator } from './storybook-configuration';

// nested code imports graph from the repo, which might have inaccurate graph version
jest.mock('nx/src/project-graph/project-graph', () => ({
  ...jest.requireActual<any>('nx/src/project-graph/project-graph'),
  createProjectGraphAsync: jest
    .fn()
    .mockImplementation(async () => ({ nodes: {}, dependencies: {} })),
}));

function listFiles(tree: Tree): string[] {
  const files = new Set<string>();
  tree.listChanges().forEach((change) => {
    if (change.type !== 'DELETE') {
      files.add(change.path);
    }
  });

  return Array.from(files).sort((a, b) => a.localeCompare(b));
}

describe('StorybookConfiguration generator', () => {
  let tree: Tree;
  const libName = 'test-ui-lib';

  beforeEach(async () => {
    tree = await createStorybookTestWorkspaceForLib(libName);

    jest.resetModules();
  });

  it('should only configure storybook', async () => {
    await storybookConfigurationGenerator(tree, <StorybookConfigurationOptions>{
      project: libName,
      generateStories: false,
      skipFormat: true,
    });

    expect(tree.exists('test-ui-lib/.storybook/main.ts')).toBeTruthy();
    expect(tree.exists('test-ui-lib/.storybook/tsconfig.json')).toBeTruthy();
    expect(
      tree.exists(
        'test-ui-lib/src/lib/test-button/test-button.component.stories.ts'
      )
    ).toBeFalsy();
    expect(
      tree.exists(
        'test-ui-lib/src/lib/test-other/test-other.component.stories.ts'
      )
    ).toBeFalsy();
  });

  it('should configure storybook to use webpack 5', async () => {
    await storybookConfigurationGenerator(tree, {
      project: libName,
      generateStories: false,
      linter: Linter.None,
      skipFormat: true,
    });

    expect(
      tree.read('test-ui-lib/.storybook/main.ts', 'utf-8')
    ).toMatchSnapshot();
  });

  it('should configure storybook with interaction tests and install dependencies', async () => {
    await storybookConfigurationGenerator(tree, <StorybookConfigurationOptions>{
      project: libName,
      generateStories: true,
    });

    expect(tree.exists('test-ui-lib/.storybook/main.ts')).toBeTruthy();
    expect(tree.exists('test-ui-lib/.storybook/tsconfig.json')).toBeTruthy();
    expect(
      tree.read(
        'test-ui-lib/src/lib/test-button/test-button.component.stories.ts',
        'utf-8'
      )
    ).toMatchSnapshot();
    expect(
      tree.read(
        'test-ui-lib/src/lib/test-other/test-other.component.stories.ts',
        'utf-8'
      )
    ).toMatchSnapshot();

    const packageJson = JSON.parse(tree.read('package.json', 'utf-8'));
    expect(packageJson.devDependencies['@storybook/angular']).toBeDefined();
    expect(
      packageJson.devDependencies['@storybook/addon-interactions']
    ).toBeDefined();
    expect(packageJson.devDependencies['@storybook/test-runner']).toBeDefined();
    expect(
      packageJson.devDependencies['@storybook/testing-library']
    ).toBeDefined();
  });

  it('should generate the right files', async () => {
    // add standalone component
    await componentGenerator(tree, {
      name: 'standalone',
      path: `${libName}/src/lib/standalone/standalone`,
      standalone: true,
      skipFormat: true,
    });
    // add secondary entrypoint
    writeJson(tree, `${libName}/package.json`, { name: libName });
    await librarySecondaryEntryPointGenerator(tree, {
      library: libName,
      name: 'secondary-entry-point',
      skipFormat: true,
    });
    // add a regular component to the secondary entrypoint
    await componentGenerator(tree, {
      name: 'secondary-button',
      path: `${libName}/secondary-entry-point/src/lib/secondary-button/secondary-button`,
      export: true,
      skipFormat: true,
    });
    // add a standalone component to the secondary entrypoint
    await componentGenerator(tree, {
      name: 'secondary-standalone',
      path: `${libName}/secondary-entry-point/src/lib/secondary-standalone/secondary-standalone`,
      standalone: true,
      export: true,
      skipFormat: true,
    });

    await storybookConfigurationGenerator(tree, <StorybookConfigurationOptions>{
      project: libName,
      generateStories: true,
      skipFormat: true,
    });

    expect(listFiles(tree)).toMatchSnapshot();
  });

  it('should generate in the correct folder', async () => {
    // add standalone component
    await componentGenerator(tree, {
      name: 'standalone',
      path: `${libName}/src/lib/standalone/standalone`,
      standalone: true,
      skipFormat: true,
    });
    // add secondary entrypoint
    writeJson(tree, `${libName}/package.json`, { name: libName });
    await librarySecondaryEntryPointGenerator(tree, {
      library: libName,
      name: 'secondary-entry-point',
      skipFormat: true,
    });
    // add a regular component to the secondary entrypoint
    await componentGenerator(tree, {
      name: 'secondary-button',
      path: `${libName}/secondary-entry-point/src/lib/secondary-button/secondary-button`,
      export: true,
      skipFormat: true,
    });
    // add a standalone component to the secondary entrypoint
    await componentGenerator(tree, {
      name: 'secondary-standalone',
      path: `${libName}/secondary-entry-point/src/lib/secondary-standalone/secondary-standalone`,
      standalone: true,
      export: true,
      skipFormat: true,
    });

    await storybookConfigurationGenerator(tree, <StorybookConfigurationOptions>{
      project: libName,
      generateStories: true,
      skipFormat: true,
    });

    expect(listFiles(tree)).toMatchSnapshot();
  });

  it('should exclude Storybook-related files from tsconfig.editor.json for applications', async () => {
    await generateTestApplication(tree, { directory: 'test-app' });

    await storybookConfigurationGenerator(tree, {
      project: 'test-app',
      generateStories: false,
      skipFormat: true,
      linter: Linter.EsLint,
    });

    const tsConfig = readJson(tree, 'test-app/tsconfig.editor.json');
    expect(tsConfig.exclude).toStrictEqual(
      expect.arrayContaining(['**/*.stories.ts', '**/*.stories.js'])
    );
  });
});
