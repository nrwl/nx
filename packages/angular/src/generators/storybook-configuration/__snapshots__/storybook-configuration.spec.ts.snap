// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StorybookConfiguration generator should configure storybook to use webpack 5 1`] = `
"import type { StorybookConfig } from '@storybook/angular';


const config: StorybookConfig = {
  stories: [
    
    '../**/*.@(mdx|stories.@(js|jsx|ts|tsx))' ],
  addons: ['@storybook/addon-essentials' , '@storybook/addon-interactions' ],
  framework: {
    name: '@storybook/angular',
    options: {
      
    },
  },
  
};

export default config;


// To customize your webpack configuration you can use the webpackFinal field.
// Check https://storybook.js.org/docs/react/builders/webpack#extending-storybooks-webpack-config
// and https://nx.dev/recipes/storybook/custom-builder-configs



"
`;

exports[`StorybookConfiguration generator should configure storybook with interaction tests and install dependencies 1`] = `
"import type { Meta, StoryObj } from '@storybook/angular';
import { TestButtonComponent } from './test-button.component';
import { within } from '@storybook/testing-library';
import { expect } from '@storybook/jest';

const meta: Meta<TestButtonComponent> = {
  component: TestButtonComponent,
  title: 'TestButtonComponent',
};
export default meta;
type Story = StoryObj<TestButtonComponent>;

export const Primary: Story = {
  args: {
    buttonType: 'button',
    style: 'default',
    age: 0,
    isOn: false,
  },
};

export const Heading: Story = {
  args: {
    buttonType: 'button',
    style: 'default',
    age: 0,
    isOn: false,
  },
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);
    expect(canvas.getByText(/test-button works!/gi)).toBeTruthy();
  },
};
"
`;

exports[`StorybookConfiguration generator should configure storybook with interaction tests and install dependencies 2`] = `
"import type { Meta, StoryObj } from '@storybook/angular';
import { TestOtherComponent } from './test-other.component';
import { within } from '@storybook/testing-library';
import { expect } from '@storybook/jest';

const meta: Meta<TestOtherComponent> = {
  component: TestOtherComponent,
  title: 'TestOtherComponent',
};
export default meta;
type Story = StoryObj<TestOtherComponent>;

export const Primary: Story = {
  args: {},
};

export const Heading: Story = {
  args: {},
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);
    expect(canvas.getByText(/test-other works!/gi)).toBeTruthy();
  },
};
"
`;

exports[`StorybookConfiguration generator should generate in the correct folder 1`] = `
[
  ".eslintignore",
  ".eslintrc.json",
  ".gitignore",
  ".prettierignore",
  ".prettierrc",
  "apps/.gitignore",
  "jest.config.ts",
  "jest.preset.js",
  "libs/.gitignore",
  "nx.json",
  "package.json",
  "test-ui-lib/.eslintrc.json",
  "test-ui-lib/.storybook/main.ts",
  "test-ui-lib/.storybook/preview.ts",
  "test-ui-lib/.storybook/tsconfig.json",
  "test-ui-lib/jest.config.ts",
  "test-ui-lib/package.json",
  "test-ui-lib/project.json",
  "test-ui-lib/README.md",
  "test-ui-lib/secondary-entry-point/ng-package.json",
  "test-ui-lib/secondary-entry-point/README.md",
  "test-ui-lib/secondary-entry-point/src/index.ts",
  "test-ui-lib/secondary-entry-point/src/lib/secondary-button/secondary-button.component.css",
  "test-ui-lib/secondary-entry-point/src/lib/secondary-button/secondary-button.component.html",
  "test-ui-lib/secondary-entry-point/src/lib/secondary-button/secondary-button.component.spec.ts",
  "test-ui-lib/secondary-entry-point/src/lib/secondary-button/secondary-button.component.stories.ts",
  "test-ui-lib/secondary-entry-point/src/lib/secondary-button/secondary-button.component.ts",
  "test-ui-lib/secondary-entry-point/src/lib/secondary-entry-point.module.ts",
  "test-ui-lib/secondary-entry-point/src/lib/secondary-standalone/secondary-standalone.component.css",
  "test-ui-lib/secondary-entry-point/src/lib/secondary-standalone/secondary-standalone.component.html",
  "test-ui-lib/secondary-entry-point/src/lib/secondary-standalone/secondary-standalone.component.spec.ts",
  "test-ui-lib/secondary-entry-point/src/lib/secondary-standalone/secondary-standalone.component.stories.ts",
  "test-ui-lib/secondary-entry-point/src/lib/secondary-standalone/secondary-standalone.component.ts",
  "test-ui-lib/src/index.ts",
  "test-ui-lib/src/lib/barrel/barrel-button/barrel-button.component.css",
  "test-ui-lib/src/lib/barrel/barrel-button/barrel-button.component.html",
  "test-ui-lib/src/lib/barrel/barrel-button/barrel-button.component.spec.ts",
  "test-ui-lib/src/lib/barrel/barrel-button/barrel-button.component.stories.ts",
  "test-ui-lib/src/lib/barrel/barrel-button/barrel-button.component.ts",
  "test-ui-lib/src/lib/barrel/barrel-button/index.ts",
  "test-ui-lib/src/lib/barrel/barrel.module.ts",
  "test-ui-lib/src/lib/nested/nested-button/nested-button.component.css",
  "test-ui-lib/src/lib/nested/nested-button/nested-button.component.html",
  "test-ui-lib/src/lib/nested/nested-button/nested-button.component.spec.ts",
  "test-ui-lib/src/lib/nested/nested-button/nested-button.component.stories.ts",
  "test-ui-lib/src/lib/nested/nested-button/nested-button.component.ts",
  "test-ui-lib/src/lib/nested/nested.module.ts",
  "test-ui-lib/src/lib/standalone/standalone.component.css",
  "test-ui-lib/src/lib/standalone/standalone.component.html",
  "test-ui-lib/src/lib/standalone/standalone.component.spec.ts",
  "test-ui-lib/src/lib/standalone/standalone.component.stories.ts",
  "test-ui-lib/src/lib/standalone/standalone.component.ts",
  "test-ui-lib/src/lib/static-member-declarations/cmp1/cmp1.component.css",
  "test-ui-lib/src/lib/static-member-declarations/cmp1/cmp1.component.html",
  "test-ui-lib/src/lib/static-member-declarations/cmp1/cmp1.component.spec.ts",
  "test-ui-lib/src/lib/static-member-declarations/cmp1/cmp1.component.stories.ts",
  "test-ui-lib/src/lib/static-member-declarations/cmp1/cmp1.component.ts",
  "test-ui-lib/src/lib/static-member-declarations/cmp2/cmp2.component.css",
  "test-ui-lib/src/lib/static-member-declarations/cmp2/cmp2.component.html",
  "test-ui-lib/src/lib/static-member-declarations/cmp2/cmp2.component.spec.ts",
  "test-ui-lib/src/lib/static-member-declarations/cmp2/cmp2.component.stories.ts",
  "test-ui-lib/src/lib/static-member-declarations/cmp2/cmp2.component.ts",
  "test-ui-lib/src/lib/static-member-declarations/static-member-declarations.module.ts",
  "test-ui-lib/src/lib/test-button/test-button.component.css",
  "test-ui-lib/src/lib/test-button/test-button.component.html",
  "test-ui-lib/src/lib/test-button/test-button.component.spec.ts",
  "test-ui-lib/src/lib/test-button/test-button.component.stories.ts",
  "test-ui-lib/src/lib/test-button/test-button.component.ts",
  "test-ui-lib/src/lib/test-other/test-other.component.css",
  "test-ui-lib/src/lib/test-other/test-other.component.html",
  "test-ui-lib/src/lib/test-other/test-other.component.spec.ts",
  "test-ui-lib/src/lib/test-other/test-other.component.stories.ts",
  "test-ui-lib/src/lib/test-other/test-other.component.ts",
  "test-ui-lib/src/lib/test-ui-lib.module.ts",
  "test-ui-lib/src/lib/variable-declare/variable-declare-button/variable-declare-button.component.css",
  "test-ui-lib/src/lib/variable-declare/variable-declare-button/variable-declare-button.component.html",
  "test-ui-lib/src/lib/variable-declare/variable-declare-button/variable-declare-button.component.spec.ts",
  "test-ui-lib/src/lib/variable-declare/variable-declare-button/variable-declare-button.component.stories.ts",
  "test-ui-lib/src/lib/variable-declare/variable-declare-button/variable-declare-button.component.ts",
  "test-ui-lib/src/lib/variable-declare/variable-declare-view/variable-declare-view.component.css",
  "test-ui-lib/src/lib/variable-declare/variable-declare-view/variable-declare-view.component.html",
  "test-ui-lib/src/lib/variable-declare/variable-declare-view/variable-declare-view.component.spec.ts",
  "test-ui-lib/src/lib/variable-declare/variable-declare-view/variable-declare-view.component.stories.ts",
  "test-ui-lib/src/lib/variable-declare/variable-declare-view/variable-declare-view.component.ts",
  "test-ui-lib/src/lib/variable-declare/variable-declare.module.ts",
  "test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-anotherview/variable-spread-declare-anotherview.component.css",
  "test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-anotherview/variable-spread-declare-anotherview.component.html",
  "test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-anotherview/variable-spread-declare-anotherview.component.spec.ts",
  "test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-anotherview/variable-spread-declare-anotherview.component.stories.ts",
  "test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-anotherview/variable-spread-declare-anotherview.component.ts",
  "test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-button/variable-spread-declare-button.component.css",
  "test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-button/variable-spread-declare-button.component.html",
  "test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-button/variable-spread-declare-button.component.spec.ts",
  "test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-button/variable-spread-declare-button.component.stories.ts",
  "test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-button/variable-spread-declare-button.component.ts",
  "test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-view/variable-spread-declare-view.component.css",
  "test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-view/variable-spread-declare-view.component.html",
  "test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-view/variable-spread-declare-view.component.spec.ts",
  "test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-view/variable-spread-declare-view.component.stories.ts",
  "test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-view/variable-spread-declare-view.component.ts",
  "test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare.module.ts",
  "test-ui-lib/src/test-setup.ts",
  "test-ui-lib/tsconfig.json",
  "test-ui-lib/tsconfig.lib.json",
  "test-ui-lib/tsconfig.spec.json",
  "tsconfig.base.json",
]
`;

exports[`StorybookConfiguration generator should generate the right files 1`] = `
[
  ".eslintignore",
  ".eslintrc.json",
  ".gitignore",
  ".prettierignore",
  ".prettierrc",
  "apps/.gitignore",
  "jest.config.ts",
  "jest.preset.js",
  "libs/.gitignore",
  "nx.json",
  "package.json",
  "test-ui-lib/.eslintrc.json",
  "test-ui-lib/.storybook/main.ts",
  "test-ui-lib/.storybook/preview.ts",
  "test-ui-lib/.storybook/tsconfig.json",
  "test-ui-lib/jest.config.ts",
  "test-ui-lib/package.json",
  "test-ui-lib/project.json",
  "test-ui-lib/README.md",
  "test-ui-lib/secondary-entry-point/ng-package.json",
  "test-ui-lib/secondary-entry-point/README.md",
  "test-ui-lib/secondary-entry-point/src/index.ts",
  "test-ui-lib/secondary-entry-point/src/lib/secondary-button/secondary-button.component.css",
  "test-ui-lib/secondary-entry-point/src/lib/secondary-button/secondary-button.component.html",
  "test-ui-lib/secondary-entry-point/src/lib/secondary-button/secondary-button.component.spec.ts",
  "test-ui-lib/secondary-entry-point/src/lib/secondary-button/secondary-button.component.stories.ts",
  "test-ui-lib/secondary-entry-point/src/lib/secondary-button/secondary-button.component.ts",
  "test-ui-lib/secondary-entry-point/src/lib/secondary-entry-point.module.ts",
  "test-ui-lib/secondary-entry-point/src/lib/secondary-standalone/secondary-standalone.component.css",
  "test-ui-lib/secondary-entry-point/src/lib/secondary-standalone/secondary-standalone.component.html",
  "test-ui-lib/secondary-entry-point/src/lib/secondary-standalone/secondary-standalone.component.spec.ts",
  "test-ui-lib/secondary-entry-point/src/lib/secondary-standalone/secondary-standalone.component.stories.ts",
  "test-ui-lib/secondary-entry-point/src/lib/secondary-standalone/secondary-standalone.component.ts",
  "test-ui-lib/src/index.ts",
  "test-ui-lib/src/lib/barrel/barrel-button/barrel-button.component.css",
  "test-ui-lib/src/lib/barrel/barrel-button/barrel-button.component.html",
  "test-ui-lib/src/lib/barrel/barrel-button/barrel-button.component.spec.ts",
  "test-ui-lib/src/lib/barrel/barrel-button/barrel-button.component.stories.ts",
  "test-ui-lib/src/lib/barrel/barrel-button/barrel-button.component.ts",
  "test-ui-lib/src/lib/barrel/barrel-button/index.ts",
  "test-ui-lib/src/lib/barrel/barrel.module.ts",
  "test-ui-lib/src/lib/nested/nested-button/nested-button.component.css",
  "test-ui-lib/src/lib/nested/nested-button/nested-button.component.html",
  "test-ui-lib/src/lib/nested/nested-button/nested-button.component.spec.ts",
  "test-ui-lib/src/lib/nested/nested-button/nested-button.component.stories.ts",
  "test-ui-lib/src/lib/nested/nested-button/nested-button.component.ts",
  "test-ui-lib/src/lib/nested/nested.module.ts",
  "test-ui-lib/src/lib/standalone/standalone.component.css",
  "test-ui-lib/src/lib/standalone/standalone.component.html",
  "test-ui-lib/src/lib/standalone/standalone.component.spec.ts",
  "test-ui-lib/src/lib/standalone/standalone.component.stories.ts",
  "test-ui-lib/src/lib/standalone/standalone.component.ts",
  "test-ui-lib/src/lib/static-member-declarations/cmp1/cmp1.component.css",
  "test-ui-lib/src/lib/static-member-declarations/cmp1/cmp1.component.html",
  "test-ui-lib/src/lib/static-member-declarations/cmp1/cmp1.component.spec.ts",
  "test-ui-lib/src/lib/static-member-declarations/cmp1/cmp1.component.stories.ts",
  "test-ui-lib/src/lib/static-member-declarations/cmp1/cmp1.component.ts",
  "test-ui-lib/src/lib/static-member-declarations/cmp2/cmp2.component.css",
  "test-ui-lib/src/lib/static-member-declarations/cmp2/cmp2.component.html",
  "test-ui-lib/src/lib/static-member-declarations/cmp2/cmp2.component.spec.ts",
  "test-ui-lib/src/lib/static-member-declarations/cmp2/cmp2.component.stories.ts",
  "test-ui-lib/src/lib/static-member-declarations/cmp2/cmp2.component.ts",
  "test-ui-lib/src/lib/static-member-declarations/static-member-declarations.module.ts",
  "test-ui-lib/src/lib/test-button/test-button.component.css",
  "test-ui-lib/src/lib/test-button/test-button.component.html",
  "test-ui-lib/src/lib/test-button/test-button.component.spec.ts",
  "test-ui-lib/src/lib/test-button/test-button.component.stories.ts",
  "test-ui-lib/src/lib/test-button/test-button.component.ts",
  "test-ui-lib/src/lib/test-other/test-other.component.css",
  "test-ui-lib/src/lib/test-other/test-other.component.html",
  "test-ui-lib/src/lib/test-other/test-other.component.spec.ts",
  "test-ui-lib/src/lib/test-other/test-other.component.stories.ts",
  "test-ui-lib/src/lib/test-other/test-other.component.ts",
  "test-ui-lib/src/lib/test-ui-lib.module.ts",
  "test-ui-lib/src/lib/variable-declare/variable-declare-button/variable-declare-button.component.css",
  "test-ui-lib/src/lib/variable-declare/variable-declare-button/variable-declare-button.component.html",
  "test-ui-lib/src/lib/variable-declare/variable-declare-button/variable-declare-button.component.spec.ts",
  "test-ui-lib/src/lib/variable-declare/variable-declare-button/variable-declare-button.component.stories.ts",
  "test-ui-lib/src/lib/variable-declare/variable-declare-button/variable-declare-button.component.ts",
  "test-ui-lib/src/lib/variable-declare/variable-declare-view/variable-declare-view.component.css",
  "test-ui-lib/src/lib/variable-declare/variable-declare-view/variable-declare-view.component.html",
  "test-ui-lib/src/lib/variable-declare/variable-declare-view/variable-declare-view.component.spec.ts",
  "test-ui-lib/src/lib/variable-declare/variable-declare-view/variable-declare-view.component.stories.ts",
  "test-ui-lib/src/lib/variable-declare/variable-declare-view/variable-declare-view.component.ts",
  "test-ui-lib/src/lib/variable-declare/variable-declare.module.ts",
  "test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-anotherview/variable-spread-declare-anotherview.component.css",
  "test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-anotherview/variable-spread-declare-anotherview.component.html",
  "test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-anotherview/variable-spread-declare-anotherview.component.spec.ts",
  "test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-anotherview/variable-spread-declare-anotherview.component.stories.ts",
  "test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-anotherview/variable-spread-declare-anotherview.component.ts",
  "test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-button/variable-spread-declare-button.component.css",
  "test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-button/variable-spread-declare-button.component.html",
  "test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-button/variable-spread-declare-button.component.spec.ts",
  "test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-button/variable-spread-declare-button.component.stories.ts",
  "test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-button/variable-spread-declare-button.component.ts",
  "test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-view/variable-spread-declare-view.component.css",
  "test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-view/variable-spread-declare-view.component.html",
  "test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-view/variable-spread-declare-view.component.spec.ts",
  "test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-view/variable-spread-declare-view.component.stories.ts",
  "test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-view/variable-spread-declare-view.component.ts",
  "test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare.module.ts",
  "test-ui-lib/src/test-setup.ts",
  "test-ui-lib/tsconfig.json",
  "test-ui-lib/tsconfig.lib.json",
  "test-ui-lib/tsconfig.spec.json",
  "tsconfig.base.json",
]
`;
